src/Microsoft.Restier.WebApi/WebApiContext.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Net.Http;
+
+namespace Microsoft.Restier.WebApi
+{
+    internal class WebApiContext
+    {
+        public HttpRequestMessage Request { get; set; }
+
+        public bool? QueryIncludeTotalCount { get; set; }
+    }
+}
```

Comment:
In general, I prefer we do not introduce a new class with only one additional properties, can we just use request.Properties to store one more attribute to indicate need total count returned?